frontend/public/components/environment.jsx

```diff
@@ -421,14 +433,15 @@ export const EnvironmentPage = connect(stateToProps)(
             <div className="co-toolbar__item">{containerDropdown}</div>
           </div>
           }
-          <div className="co-m-pane__body-group">
-            <h3 className="co-section-heading-tertiary">Single values (env)
+          <div className={classNames({'co-m-pane__body-group': !currentEnvVars.isCreate})}>
+            { !currentEnvVars.isCreate && <h3 className="co-section-heading-tertiary">Single values (env)
               {
                 !readOnly && <FieldLevelHelp content={
                   <div>Define environment variables as key-value pairs to store configuration settings. You can enter text or add values from a ConfigMap or Secret. Drag and drop environment variables to change the order in which they are run. A variable can reference any other variables that come before it in the list, for example <code>FULLDOMAIN = $(SUBDOMAIN).example.com</code>.</div>} />
               }
             </h3>
-            <NameValueEditorComponent nameValueId={containerIndex} nameValuePairs={envVar[EnvType.ENV]} updateParentData={this.updateEnvVars} addString="Add Value" nameString="Name" readOnly={readOnly} allowSorting={true} configMaps={configMaps} secrets={secrets} />
+            }
+            <NameValueEditorComponent nameValueId={containerIndex} nameValuePairs={envVar[EnvType.ENV]} updateParentData={this.updateEnvVars} addString="Add Value" nameString="Name" readOnly={readOnly} allowSorting={true} configMaps={configMaps} secrets={secrets} addConfigMapSecret={!currentEnvVars.isCreate} />
```

Comment:
> `addConfigMapSecret={!currentEnvVars.isCreate}`

I'd also make this a `addConfigMapSecret` prop passed to the `EnvironmentPage` component. I think this should be set by the parent component without making assumptions about context here.